tests/test_coastal_vulnerability.py

```diff
@@ -713,26 +713,67 @@ def test_complete_run(self):
-        pandas.testing.assert_series_equal(
-            actual_values_df['shore_id'], habitat_csv['shore_id'])
+        results_vector = gdal.OpenEx(
+            os.path.join(args['workspace_dir'], 'coastal_exposure.gpkg'),
+            gdal.OF_VECTOR)
+        results_layer = results_vector.GetLayer()
+
+        # shore points aren't necessarily created/numbered in the same order,
+        # so we can't compare by the fid or shore_id fields
+        # instead check that each point has the expected attributes
+        fields = [
+            'R_hab', 'R_wind', 'R_wave', 'R_surge', 'R_relief', 'R_geomorph',
+            'R_slr', 'population', 'exposure', 'habitat_role',
+            'exposure_no_habitats']
+        # mapping of x, y point coords to their fields in the order above
+        expected_point_results = {
+            (265195.9118871244, 5465342.359978485): [
+                4.05, 5, 5, 1, 5, 1, 5, 0, 3.063308387, 0.0936167899, 3.1569251777],
+            (275802.584962168, 5469283.47691652): [
+                5, 3, 1, 1, 5, 1, 4, 0, 2.2587827631, 0, 2.2587827631],
+            (376479.9819538344, 5383636.197270025): [
+                4.05, 5, 5, 4, 4, 4, 3, numpy.nan, 4.0989337064, 0.125266204816, 4.2241999112],
+            (355807.04065901926, 5394736.771414153): [
+                5, 5, 4, 4, 3, 4, 2, 0, 3.70591872429, 0, 3.70591872429],
+            (344736.25795214524, 5411347.428706937): [
+                5, 2, 4, 3, 3, 4, 1, 0, 2.8261463109, 0, 2.8261463109],
+            (361290.81087879254, 5427975.474574804): [
+                5, 1, 3, 5, 1, 4, 2, 0, 2.493898362454, 0, 2.493898362454],
+            (368269.9048903524, 5449541.99543876): [
+                5, 1, 1, 5, 2, 4, 2, 0.008, 2.3535468936, 0, 2.3535468936],
+            (361341.1463245464, 5433678.995435326): [
+                5, 2, 1, 5, 1, 4, 1, 0, 2.1316631165, 0, 2.1316631165],
+            (339388.60793905176, 5428895.077843302): [
+                5, 3, 4, 4, 2, 4, 1, 0, 2.94471340036, 0, 2.94471340036],
+            (318255.8192637532, 5423278.964182078): [
+                5, 4, 3, 2, 3, 2.5, 1, 0, 2.6426195539, 0, 2.6426195539],
+            (300130.0010361069, 5437435.4230010025): [
+                1.7999999, 4, 5, 2, 5, 1, 3, 0, 2.712353253, 0.426215219066, 3.138568472],
+            (306112.5574508078, 5450095.8865171345): [
+                5, 1, 3, 3, 2, 1, 3, 0, 2.225039271, 0, 2.225039271],
+            (290844.10673833836, 5460752.369983306): [
+                5, 4, 2, 2, 2, 1, 4, 0, 2.5169979012, 0, 2.5169979012],
+            (286971.1600469994, 5475571.525391179): [
+                5, 1, 1, 3, 1, 1, 4, 0, 1.7948229213, 0, 1.7948229213],
+            (268404.5296770451, 5479539.256472221): [
+                5, 2, 2, 1, 1, 1, 5, 0, 1.9306977288, 0, 1.9306977288],
+            (252692.2222200262, 5482540.950248547): [
+                5, 3, 2, 1, 4, 1, 5, 0, 2.49389836245, 0, 2.49389836245]
+        }

```

Comment:
I like having the expected data hardcoded!

Reply:
Out of curiosity, did you notice if these coordinates match the previous regression data? Or did the coords change slightly along with the order of the points? It doesn't really matter either way I don't think.

Reply:
They looked to be identical.